src/gui/src/heatMap.cpp

```diff
@@ -88,6 +84,41 @@ void HeatMapDataSource::registerHeatMap()
   Gui::get()->registerHeatMap(this);
 }
 
+void HeatMapDataSource::dumpToFile(const std::string& file)
```

Comment:
Can this be a const method?  Similarly with its callers

Reply:
It's calling `ensureMap` which populates the map with data, so I don't think it could be at the moment.